server/utils/parseMentions.ts

```diff
@@ -0,0 +1,32 @@
+import { Node } from "prosemirror-model";
+import { Document, Revision } from "@server/models";
+import DocumentHelper from "@server/models/helpers/DocumentHelper";
+
+/**
+ * Parse a list of mentions contained in a document or revision
+ *
+ * @param document Document or Revision
+ * @returns An array of mentions in passed document or revision
+ */
+export default function parseMentions(
+  document: Document | Revision
+): Record<string, string>[] {
+  const node = DocumentHelper.toProsemirror(document);
+  const mentions: Set<Record<string, string>> = new Set();
```

Comment:
Is set achieving anything now? Could probably just be an array as we convert it to one at the end